waku/factory/node_factory.nim

```diff
@@ -226,16 +231,43 @@ proc setupProtocols(
     except CatchableError:
       return err("failed to mount waku RLN relay protocol: " & getCurrentExceptionMsg())
 
-  if conf.store:
-    # Archive setup
-    let archiveDriverRes = waitFor ArchiveDriver.new(
+  if conf.store and conf.legacyStore:
+    let archiveDriverRes = waitFor legacy_driver.ArchiveDriver.new(
+      conf.storeMessageDbUrl, conf.storeMessageDbVacuum, conf.storeMessageDbMigration,
+      conf.storeMaxNumDbConnections, onFatalErrorAction,
+    )
+    if archiveDriverRes.isErr():
+      return err("failed to setup legacy archive driver: " & archiveDriverRes.error)
+
+    let retPolicyRes =
+      legacy_policy.RetentionPolicy.new(conf.storeMessageRetentionPolicy)
+    if retPolicyRes.isErr():
+      return err("failed to create retention policy: " & retPolicyRes.error)
+
+    let mountArcRes =
+      node.mountLegacyArchive(archiveDriverRes.get(), retPolicyRes.get())
+    if mountArcRes.isErr():
+      return err("failed to mount waku legacy archive protocol: " & mountArcRes.error)
+
+    # Store setup
+    let rateLimitSetting: RateLimitSetting =
+      (conf.requestRateLimit, chronos.seconds(conf.requestRatePeriod))
```

Comment:
Do we need to duplicate all this code? Can't we reuse it and just change the creation of the driver and the mounting of the protocol based on which one was configured?

Reply:
I just though it would be easier to just remove this code block once we no longer support v2. Should I change that?